tests/gabbits/create-snippet-errors.yaml

```diff
@@ -0,0 +1,59 @@
+fixtures:
+  - XSnippetApi
+
+tests:
+  - name: create a new snippet (invalid method)
+    PUT: /v1/snippets
+    request_headers:
+      content-type: application/json
+    data:
+      title: Hello, World!
+      syntax: python
+      content: print('Hello, World!')
+      tags:
+        - spam
+        - eggs
+    status: 404
+
+  - name: create a new snippet (invalid JSON)
+    POST: /v1/snippets
+    request_headers:
+      content-type: application/json
+    data: spam
+    status: 400
```

Comment:
Just curious, what would be the error message? Taking into account that this is a part of normal flow (someone misspelled `content` or something), shouldn't we test the error message?

Reply:
Yeah, that's a good point. At the moment, it's not pretty :( Because it's now we who produce this error message, it's not reported as all other errors that we control: instead of HTTP status + `{"message": "..."}` in the response body, you will see some text encoded as HTML.

Reply:
Do you plan to address this later on? I wonder, should we track our planned work in issue tracker.

Reply:
+1 for using the issue tracker. For this one specifically, I was going to fix your comment before we merge this PR. I'll ping you when it's ready for review!